### PR TITLE
"prepareBuildings","prepareRoads", "prepareRails" processes creation

### PR DESCRIPTION
**Pull request by ELSW56**
Creation of "prepareBuilding", "prepareRoad" and "prepareRail" processes, their test and their test resources.
Modification of the methods implied in the processes
Modification of the "prepareBuilding" process input names

### CODE REVIEW

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy [hunk 1]
```diff
@@ -19,11 +20,43 @@ import org.orbisgis.processmanagerapi.IProcess
 
 @BaseScript PrepareData prepareData
 
+//TODO : use stream api instead of file
```
Comment (ebocher) [inline prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: Add javadoc

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy [hunk 2]
```diff
@@ -19,11 +20,43 @@ import org.orbisgis.processmanagerapi.IProcess
...
+
+static IProcess prepareBuildings() {
+    return processFactory.create(
+            "Prepare the building layer with OSM data",
+            [datasource   : JdbcDataSource,
+             prefix : String,
```
Comment (ebocher) [inline prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: tablePrefix

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy [hunk 3]
```diff
@@ -19,11 +20,43 @@ import org.orbisgis.processmanagerapi.IProcess
...
+             inputOptions: Map,
```
Comment (ebocher) [inline prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: ouputColumnNames

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy [hunk 4]
```diff
@@ -19,11 +20,43 @@ import org.orbisgis.processmanagerapi.IProcess
...
+             inputTagKeys: String[],
```
Comment (ebocher) [inline prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: tagKeys

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy [hunk 5]
```diff
@@ -19,11 +20,43 @@ import org.orbisgis.processmanagerapi.IProcess
...
+             inputTagValues: String[],
```
Comment (ebocher) [inline prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: tagValues

File: prepare_data/src/test/groovy/org/orbisgis/osm/OSMTests.groovy [hunk 6]
```diff
@@ -10,9 +10,31 @@ class OSMTests {
...
      */
     @Test
-    void loadH2GIS() {
-        def h2GIS = H2GIS.open([databaseName: './target/loadH2GIS'])
-        assertNotNull(h2GIS)
+    void prepareBuildingsTest() {
+        def h2GIS = H2GIS.open('./target/buildingdb')
+        h2GIS.load(new File(this.class.getResource("zoneExtended.osm").toURI()).getAbsolutePath(),"ext",true)
+        assertNotNull(h2GIS.getTable("EXT_NODE"))
+        println('Load OSM tables OK')
```
Comment (ebocher) [inline prepare_data/src/test/groovy/org/orbisgis/osm/OSMTests.groovy]: logger

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy [hunk 7]
```diff
@@ -19,11 +20,43 @@ import org.orbisgis.processmanagerapi.IProcess
...
+             buildingTableName: String,
+             zoneBufferTableName: String],
```
Comment (ebocher) [inline prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: filteringZoneTableName

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy (hunk 1 above)
Comment (ELSW56) [reply prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: Done

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy (hunk 2 above)
Comment (ELSW56) [reply prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: Done

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy (hunk 3 above)
Comment (ELSW56) [reply prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: Done

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy (hunk 4 above)
Comment (ELSW56) [reply prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: Done

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy (hunk 5 above)
Comment (ELSW56) [reply prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: Done

File: prepare_data/src/test/groovy/org/orbisgis/osm/OSMTests.groovy (hunk 6 above)
Comment (ELSW56) [reply prepare_data/src/test/groovy/org/orbisgis/osm/OSMTests.groovy]: Done

File: prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy (hunk 7 above)
Comment (ELSW56) [reply prepare_data/src/main/groovy/org/orbisgis/osm/OSMGISLayers.groovy]: Done

**Pull request merged**